components/identities/actions/addEvmAddrModal.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  CircularProgress,
+  InputAdornment,
+  Modal,
+  TextField,
+} from "@mui/material";
+import { useContractWrite } from "@starknet-react/core";
+import React, { FunctionComponent, useEffect, useRef, useState } from "react";
+import styles from "../../../styles/components/evmModalMessage.module.css";
+import Button from "../../UI/button";
+import { hexToDecimal } from "../../../utils/feltService";
+import ConfirmationTx from "../../UI/confirmationTx";
+import { useNotificationManager } from "../../../hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "../../../utils/constants";
+import { Identity } from "../../../utils/apiWrappers/identity";
+import identityChangeCalls from "../../../utils/callData/identityChangeCalls";
+import { shortString } from "starknet";
+import { isValidEns } from "@/utils/ensService";
+import { ethers } from "ethers";
+
+type AddEvmAddrModalProps = {
+  handleClose: (showNotif: boolean) => void;
+  isModalOpen: boolean;
+  identity?: Identity;
+};
+
+const AddEvmAddrModal: FunctionComponent<AddEvmAddrModalProps> = ({
+  handleClose,
+  isModalOpen,
+  identity,
+}) => {
+  const [evmAddress, setEvmAddress] = useState<string | undefined>(
+    identity?.evmAddress
+  );
+  const [fieldInput, setFieldInput] = useState<string>(evmAddress ?? "");
+  const { addTransaction } = useNotificationManager();
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+  const [message, setMessage] = useState<string | undefined>(undefined);
+  const [loading, setLoading] = useState(false);
+  const abortControllerRef = useRef<AbortController | null>(null);
+
```

Comment:
> Ensure that the `identity` prop is always defined before accessing its properties to avoid potential runtime errors.
> 
> ```diff
> - const [evmAddress, setEvmAddress] = useState<string | undefined>(identity?.evmAddress);
> + const [evmAddress, setEvmAddress] = useState<string | undefined>(identity ? identity.evmAddress : undefined);
> ```
> 
> Committable suggestion

Isn't it the purpose of using the `?.` ?